### PR TITLE
Parse config in factory function

### PR DESCRIPTION
**Pull request by MartinForReal**
We can remove unnecessary strings. Index calls by split host and port when instance is created.

### CODE REVIEW

File: armbalancer.go [hunk 1]
```diff
@@ -82,8 +85,17 @@ type recyclableTransport struct {
 func newRecyclableTransport(id int, parent *http.Transport, host string, recycleThreshold, minReqsBeforeRecycle int64) *recyclableTransport {
 	tx := parent.Clone()
 	tx.MaxConnsPerHost = 1
+	hostUrl, err := url.Parse("https://" + host)
```
Comment (jveski) [inline armbalancer.go:88]: `newRecyclableTransport` is invoked for each pooled transport - let's move the URL parsing up the stack to `New`.

Comment (jveski) [inline armbalancer.go:88]: Please use `net.SplitHostPort`  to avoid prepending `"https://"`

Comment (MartinForReal) [reply armbalancer.go]: fixed in new commit

Comment (MartinForReal) [reply armbalancer.go]: fixed in new commit

File: armbalancer.go [hunk 2]
```diff
@@ -42,6 +44,13 @@ func New(opts Options) http.RoundTripper {
 	if opts.Host == "" {
 		opts.Host = "management.azure.com"
 	}
+	host, port, err := net.SplitHostPort(opts.Host)
```
Comment (jveski) [inline armbalancer.go:47]: Please add a unit test to prove that this code behaves as expected when a port is specified, not specified, or some invalid host is provided.

Comment (MartinForReal) [reply armbalancer.go:54]: added some ut cases. PTAL. Thanks!

**Review by jveski — APPROVED** [review]
:ship:

**Pull request merged**